lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2039 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+
+module: cloudfront_distribution
+
+short_description: create, update and delete aws cloudfront distributions.
+
+description:
+    - Allows for easy creation, updating and deletion of CloudFront
+      distributions.
+
+requirements:
+  - boto3 >= 1.0.0
+  - python >= 2.6
+
+version_added: "2.4"
+
+author: Willem van Ketwich (@wilvk)
+
+options:
+
+    state:
+      description:
+        - The desired state of the distribution or streaming distribution.
+          present - creates a new distribution or updates an existing
+            distribution.
+          absent - deletes an existing distribution.
+      choices: ['present', 'absent']
+      required: false
```

Comment:
Thanks for the feedback - there is extra functionality in this module for generating presigned urls that prevents the `state `parameter from being required - ie. using `generate_presigned_url` (see example on line 577).